system-contracts/contracts/erc4337/EntryPointV01.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IEntryPoint, PackedUserOperation} from "../interfaces/IEntryPoint.sol";
+import {IBootloaderUtilities} from "../interfaces/IBootloaderUtilities.sol";
+import {IAccount} from "../interfaces/IAccount.sol";
+import {IAccountCodeStorage} from "../interfaces/IAccountCodeStorage.sol";
+import {INonceHolder} from "../interfaces/INonceHolder.sol";
+import {IContractDeployer} from "../interfaces/IContractDeployer.sol";
+import {ISystemContext} from "../interfaces/ISystemContext.sol";
+import {ContractDeployer} from "../ContractDeployer.sol";
+import {Transaction, TransactionHelper, EIP_712_TX_TYPE, LEGACY_TX_TYPE, EIP_2930_TX_TYPE, EIP_1559_TX_TYPE} from "../libraries/TransactionHelper.sol";
+import {BOOTLOADER_FORMAL_ADDRESS, SYSTEM_CONTEXT_CONTRACT, NONCE_HOLDER_SYSTEM_CONTRACT, ACCOUNT_CODE_STORAGE_SYSTEM_CONTRACT, DEPLOYER_SYSTEM_CONTRACT} from "../Constants.sol";
+import {SystemContractBase} from "../abstract/SystemContractBase.sol";
+import {SystemContext} from "../SystemContext.sol";
+import {RLPEncoder} from "../libraries/RLPEncoder.sol";
+import {EfficientCall} from "../libraries/EfficientCall.sol";
+import {UnsupportedTxType, InvalidSig, SigField, HashMismatch} from "../SystemContractErrors.sol";
+
+import {SystemContractsCaller, CalldataForwardingMode} from "../libraries/SystemContractsCaller.sol";
+
+/**
+ * @author Matter Labs
+ * @custom:security-contact security@matterlabs.dev
+ * @notice An EIP4337 EntryPoint contract implementation
+ * built on top of the native ZKsync Account Abstraction.
+ */
+contract EntryPointV01 is IEntryPoint, SystemContractBase {
+    using TransactionHelper for *;
+
+    function handleUserOps(PackedUserOperation[] calldata _ops) external {
```

Comment:
to not forget: as of now, the function will work for EOAs as well, I dont think this is what we want (the ability to start tx out of an empty account inside the transaction